pkg/util/error_list.go

```diff
@@ -27,10 +27,10 @@ import (
 
 // This helper implements the error interface for []error, but prevents
 // accidental conversion of []error to error.
-type errorList []error
+type ErrorList []error
```

Comment:
I don't think filter is explicitly on lists though - it's a statement of "if this error represents any of these types, ignore it".  Since []error is an aggregate error, if 2 errors in that slice match, but the third doesn't, the result should be errs[2].  If it's not an aggregate, it's either the identify error or nil.  If we have a true hierarchal error, it's the responsibility of the matcher to define that.  I'm ok with errlist package, and if this is structured as an interface cast it's slightly clearer.


> On Jan 4, 2015, at 1:40 AM, Tim Hockin <notifications@github.com> wrote:
> 
> In pkg/util/error_list.go:
> 
> > @@ -27,10 +27,10 @@ import (
> >  
> >  // This helper implements the error interface for []error, but prevents
> >  // accidental conversion of []error to error.
> > -type errorList []error
> > +type ErrorList []error
> Having read over just this portion of the PR< I think we want a different strategy.
> 
> What about something like:
> 
> 1) break the existing SliceToError() stuff into pkg/util/errlist (and rename to errlist.ToError(errs)).
> 2) add your logic as errlist.FilterOut(errs []error, funcs ...MatchFunc) there
> 
> Then we can argue about flattening...
> 
> —
> Reply to this email directly or view it on GitHub.
>